src/context/index.tsx

```diff
@@ -8,7 +9,11 @@ const AppProviders: React.FC = ({ children }) => {
         <AppContent>
             <AuthProvider>
                 <AppTheme>
-                    <AppRouter>{children}</AppRouter>
+                    <AppRouter>
+                        <ConfirmationModalContextProvider>
+                            {children}
+                        </ConfirmationModalContextProvider>
```

Comment:
Needs to be at least inside the theme... I think putting it under the router makes the most sense. That way if we ever need to do a "Are you sure you want to leave this page" kinda thing we can.